pyproject.toml

```diff
@@ -132,7 +132,10 @@ src = ["src"]
 
 [tool.ruff.lint]
 select = ["E", "F", "D"]
-ignore = ["E501"]
+ignore = [
+  "D418",  # Bans docstrings on overloads. Pylance supports these and they are useful.
```

Comment:
We could ignore it on the specific file, but given that Pylance (VSCode's content assistance plugin) uses docstrings from overloads, I thought it might accidentally discourage use in future if we didn't disable it globally (and it also is a good place to explain why we're suppressing it). Making them comments instead of docstrings will stop Pylance picking them up.

I will add in a screenshot of this information showing up in VSCode to the PR description as motivation, thanks for flagging!